From 8b34dc61ab1f46ff669936ee653515b8f6afd0b1 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Wed, 7 Oct 2020 18:15:29 -0400
Subject: [PATCH] fec: lib: Remove use of FEC_API for function definitions.

This fixes building with MSVC. Since these files first get built into an
object that is *not* a shared library, FEC_API was resolving to
__declspec(dllimport), which should never be used for function
definitions. Warnings still occur about inconsistent dll linkage because
the gr_fec_rs target is built into an object library that doesn't define
"gnuradio_fec_EXPORTS", but the end result works. These warnings have
always existed previously.
---
 gr-fec/lib/reed-solomon/decode_rs.h       | 4 +---
 gr-fec/lib/reed-solomon/decode_rs_ccsds.c | 4 +---
 gr-fec/lib/reed-solomon/encode_rs.h       | 4 +---
 gr-fec/lib/reed-solomon/encode_rs_ccsds.c | 4 +---
 4 files changed, 4 insertions(+), 12 deletions(-)

diff --git a/gr-fec/lib/reed-solomon/decode_rs.h b/gr-fec/lib/reed-solomon/decode_rs.h
index a2feeb090..1c18f0565 100644
--- a/gr-fec/lib/reed-solomon/decode_rs.h
+++ b/gr-fec/lib/reed-solomon/decode_rs.h
@@ -3,8 +3,6 @@
  * May be used under the terms of the GNU General Public License (GPL)
  */
 
-#include <gnuradio/fec/api.h>
-
 #ifdef DEBUG
 #include <stdio.h>
 #endif
@@ -17,7 +15,7 @@
 
 #define min(a, b) ((a) < (b) ? (a) : (b))
 
-FEC_API int DECODE_RS(
+int DECODE_RS(
 #ifndef FIXED
     void* p,
 #endif
diff --git a/gr-fec/lib/reed-solomon/decode_rs_ccsds.c b/gr-fec/lib/reed-solomon/decode_rs_ccsds.c
index 9196260e6..be6e2018e 100644
--- a/gr-fec/lib/reed-solomon/decode_rs_ccsds.c
+++ b/gr-fec/lib/reed-solomon/decode_rs_ccsds.c
@@ -8,9 +8,7 @@
 #include "ccsds.h"
 #include "fixed.h"
 
-#include <gnuradio/fec/api.h>
-
-FEC_API int decode_rs_ccsds(unsigned char* data, int* eras_pos, int no_eras)
+int decode_rs_ccsds(unsigned char* data, int* eras_pos, int no_eras)
 {
     int i, r;
     unsigned char cdata[NN];
diff --git a/gr-fec/lib/reed-solomon/encode_rs.h b/gr-fec/lib/reed-solomon/encode_rs.h
index 75051426d..45e1ab720 100644
--- a/gr-fec/lib/reed-solomon/encode_rs.h
+++ b/gr-fec/lib/reed-solomon/encode_rs.h
@@ -3,11 +3,9 @@
  * May be used under the terms of the GNU General Public License (GPL)
  */
 
-#include <gnuradio/fec/api.h>
-
 #include <string.h>
 
-FEC_API void ENCODE_RS(
+void ENCODE_RS(
 #ifndef FIXED
     void* p,
 #endif
diff --git a/gr-fec/lib/reed-solomon/encode_rs_ccsds.c b/gr-fec/lib/reed-solomon/encode_rs_ccsds.c
index 1ebd3a443..7080b3f41 100644
--- a/gr-fec/lib/reed-solomon/encode_rs_ccsds.c
+++ b/gr-fec/lib/reed-solomon/encode_rs_ccsds.c
@@ -8,9 +8,7 @@
 #include "ccsds.h"
 #include "fixed.h"
 
-#include <gnuradio/fec/api.h>
-
-FEC_API void encode_rs_ccsds(unsigned char* data, unsigned char* parity)
+void encode_rs_ccsds(unsigned char* data, unsigned char* parity)
 {
     int i;
     unsigned char cdata[NN - NROOTS];
-- 
2.20.1

